File: a022(palindrome)/C++/palindrome.cpp
#include <iostream>
#include <algorithm>

bool palindrome(const std::string input_string) {
  std::string input_reverse = input_string;
  std::reverse(input_reverse.begin(), input_reverse.end());
  for(int i = 0; i < input_string.size(); i++) {
    if(input_string[i] != input_reverse[i]) {
      return false;
    }
  }
  return true;
}

int main(void) {
  std::string input_string;
  while (std::cin >> input_string) {
    if(palindrome(input_string)) {
      std::cout << "yes" << '\n';
    } else {
      std::cout << "no" << '\n';
    }
  }
  return 0;
}

File: a022(palindrome)/Python/palindrome.py

def palindrome(input_string: list) -> bool:
    input_reverse = input_string.copy()
    input_reverse.reverse()
    for i in range(len(input_string)):
        if input_string[i] != input_reverse[i]:
            return False
    return True

while True:
    try:
        input_string = list(input())
        if palindrome(input_string):
            print('yes')
        else:
            print('no')
    except:
        break
